content/docs/getting-started/futures.md

```diff
@@ -163,27 +163,22 @@ The `Future` trait is as follows:
 ```rust,ignore
 trait Future {
     /// The type of the value returned when the future completes.
-    type Item;
-
-    /// The type representing errors that occurred while processing the computation.
-    type Error;
+    type Output;
 
     /// The function that will be repeatedly called to see if the future
     /// has completed or not
-    fn poll(&mut self) -> Result<Async<Self::Item>, Self::Error>;
+    fn poll(self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output>;
 }
 ```
 
-For now it's just important to know that futures have two associated types: `Item`
-and `Error`. `Item` is the type of the value that the `Future` will yield when it
-completes. `Error` is the type of Error that the `Future` may yield if there's an
-error before that prevents the `Future` from completing successfully.
+For now it's just important to know that futures have an associated type: `Output`.
+It is the type of the value that the `Future` will yield when it completes.
 
-Finally, `Future`s have one method named `poll`. We won't go into too much detail
+Additionally, `Future`s have one method named `poll`. We won't go into too much detail
 about `poll` in this section since you don't need to know about `poll` to use
 futures with combinators. The only thing to be aware for now is that `poll` is
 what the runtime will call in order to see if the `Future` is complete yet or not.
```

Comment:
A comparison I like to make that may be worth including is that the relationship between `Future::poll` and the `async`/`await` keywords is very similar to the relationship between `Iterator::next` and `for` loops: you don't need to be aware of the details of how `next` is implemented to iterate over a slice in a `for` loop. Futures and async/await are similar.